src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/CertificateStore.cs

```diff
@@ -135,22 +133,40 @@ public Errorable<X509Certificate2> FindByThumbprint(string purpose, CertificateT
         /// <param name="storeLocation">Location within the store to check.</param>
         /// <returns>An <see cref="Errorable{X509Certificate2}"/> containing a certificate if found, or an
         /// error otherwise.</returns>
-        private static X509Certificate2 FindByThumbprint(CertificateThumbprint thumbprint, StoreName storeName, StoreLocation storeLocation)
+        private static X509Certificate2 FindByThumbprint(
+            CertificateThumbprint thumbprint,
+            StoreName storeName,
+            StoreLocation storeLocation) =>
+            ReadFromStore(
+                storeName,
+                storeLocation,
+                store => store.Certificates.Find(thumbprint).SingleOrDefault(),
+                null);
```

Comment:
Why introduce this complex helper function vs just catching the relevant exceptions in both methods? 😕

Reply:
Because the bit which seemed complex or error-prone to me was knowing what exceptions to catch. If that changed (as it already has twice), I think it's best to only have to update the code it in one place. It looks as if a previous fix for the same issue only made the change in one place and missed the other.

Reply:
We've agreed to merge in PR #129 to substitute for the changes to this file.